contracts/utils/contracts/ScaleCodec.sol

```diff
@@ -19,6 +19,7 @@ library ScaleCodec {
         pure
         returns (uint256 v, uint8 m)
     {
+        require(data.length > 0, "Compact data cannot be empty");
```

Comment:
It is safe, but is a redundant check.
If the decode type is `uintN`, data.length > 0.

Reply:
> If the decode type is uintN, data.length > 0.
what's this mean?

Reply:
```
bytes data = scalec.encode(uintN);
data.length > 0
```

Reply:
If the data is uintN type, data.length > 0.

Reply:
If you want do a unitN scalec decode. the check is redundant.
This issue occur root case is that decode a Non UintN scale type.